### ISSUE TITLE
PiVPN OpenVPN Server and Connection Refused Error

### ISSUE THREAD

**Issue by condraj2** [furlongm/openvpn-monitor#165]
Hi! I'm running Raspbian Buster on a RPI4+ and having issues with final configuration. I have PiVPN via OpenVPN set up and working properly - I am able to connect to it via my mobile device as intended. I'm trying to set OpenVPN-Monitor up on the same RPI4.

I followed the virtualenv + pip + gunicorn instructions and am initializing the server with the `sudo gunicorn openvpn-monitor -b 0.0.0.0:80` command, which does bring the server up and down as expected, and does pull the configurations from `/etc/openvpn-monitor.conf` (if I make changes in that file and restart the server, they are reflected [e.g. changing the name of the site, date and time order, etc.]). `openvpn-monitor.py` is in the same directory (/etc/) as the .conf file. I do appear to have a totally separate folder for some reason with duplicate files located at `/var/www/html/openvpn/` but I am not entirely sure why. When I initialize the server it appears to only pull from those located within /etc/.

My issues are 3-fold:

**1.** Logo doesn't seem to update (by pointing it directly to an image URL online nor to a locally saved image in a directory). It still displays a broken image icon.
**2.** I can access the management interface via 127.0.0.1 on the host, but get a 404 error when attempting 127.0.0.1/openvpn-monitor/ as instructed. Error is: `Not found: '/openvpn-monitor'`
**3.** The big issue: `Could not connect to 127.0.0.1:5555 (Connection refused)`

I have tried everything I can think of to resolve this one but I just can't seem to get it to work. I have tried adding the following to **server.conf:**

```
management 127.0.0.1 5555
--management 127.0.0.1 5555
```

My **openvpn-monitor.conf** file is as follows:

```
#### OpenVPN-Monitor
# The below should help you quickly configure your vpn monitor with relevant settings

[openvpn-monitor]
site=PiVPN
logo=/home/pi/Downloads/openvpn.png
latitude=Redacted
longitude=Redacted
Maps=True
geoip_data=/var/lib/GeoIP/GeoLite2-City.mmdb
pervpn_control=True
itemtype_control=True
allowFullscreen=True
datetime_format=%m/%d/%Y %H:%M:%S

#### Per VPN Settings
# The below should help you quickly configure connections to VPNs.

[VPN1]
host=127.0.0.1
port=5555
name=Home VPN
show_disconnect=True
connection_lines=True
marker=True
```
The output of **sudo python3 openvpn-monitor.py -d** is as follows:

```
INFO: Using config file: ./openvpn-monitor.conf
DEBUG:
 === begin section
{'site': 'PiVPN', 'logo': '/home/pi/Downloads/openvpn.png', 'latitude': 'REDACTED', 'longitude': 'REDACTED', 'maps': 'True', 'geoip_data': '/var/lib/GeoIP/GeoLite2-City.mmdb', 'datetime_format': '%m/%d/%Y %H:%M:%S'}
=== end section
DEBUG:
 === begin section
{'host': '127.0.0.1', 'port': '5555', 'name': 'Home VPN', 'show_disconnect': True, 'connection_lines': 'True', 'marker': 'True'}
=== end section
WARNING: socket error: [Errno 111] Connection refused
Content-Type: text/html

<!doctype html>
<html lang="en"><head>
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1">
<title>PiVPN OpenVPN Status Monitor</title>
<meta http-equiv="refresh" content="300" />
<link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />
<link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap-theme.min.css" integrity="sha256-8uHMIn1ru0GS5KO+zf7Zccf8Uw12IA5DrdEcmMuWLFM=" crossorigin="anonymous" />
<link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/jquery.tablesorter/2.31.3/css/theme.bootstrap_3.min.css" integrity="sha256-vgjicWNWkVklkfuqKnQth9ww987V7wCOzh6A0qkJ2Lw=" crossorigin="anonymous" />
<link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/leaflet/1.6.0/leaflet.css" integrity="sha256-SHMGCYmST46SoyGgo4YR/9AlK1vf3ff84Aq9yK4hdqM=" crossorigin="anonymous" />
<link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/leaflet.fullscreen/1.6.0/Control.FullScreen.css" integrity="sha256-RTALnHN76PJ32RJx2mxggy+RUt9TIRV+mfPLSLbI75A=" crossorigin="anonymous" />
<style>
.panel-custom {
   background-color:#777;
   color:#fff;
   font-size:80%;
   vertical-align:baseline;
   padding:.4em .4em .4em;
   line-height:1;
   font-weight:700;
}
</style>
<script src="//cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script>
<script src="//cdnjs.cloudflare.com/ajax/libs/jquery.tablesorter/2.31.3/js/jquery.tablesorter.min.js" integrity="sha256-dtGH1XcAyKopMui5x20KnPxuGuSx9Rs6piJB/4Oqu6I=" crossorigin="anonymous"></script>
<script src="//cdnjs.cloudflare.com/ajax/libs/jquery.tablesorter/2.31.3/js/jquery.tablesorter.widgets.min.js" integrity="sha256-GxbszpUzg+iYIcyDGyNVLz9Y0dQvzmQgXXVk5cHJbw0=" crossorigin="anonymous"></script>
<script src="//cdnjs.cloudflare.com/ajax/libs/jquery.tablesorter/2.31.3/js/parsers/parser-network.min.js" integrity="sha256-rV5r5lDTGqNm0Nw6Q/uRKvlco8igIg2PSo3daI1vykA=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>
<script>$(document).ready(function(){
$("table.tablesorter").tablesorter({sortList: [[0,0]], theme:"bootstrap", headerTemplate:"{content} {icon}", widgets:["uitheme"]});
});</script>
<script src="//cdnjs.cloudflare.com/ajax/libs/leaflet/1.6.0/leaflet.js" integrity="sha256-fNoRrwkP2GuYPbNSJmMJOCyfRB2DhPQe0rGTgzRsyso=" crossorigin="anonymous"></script>
<script src="//cdnjs.cloudflare.com/ajax/libs/OverlappingMarkerSpiderfier-Leaflet/0.2.6/oms.min.js" integrity="sha256-t+V41b9l6j8GMYAbpcnZbib1XiYwCAsDibD8sI1D7+Y=" crossorigin="anonymous"></script>
<script src="//cdnjs.cloudflare.com/ajax/libs/leaflet.fullscreen/1.6.0/Control.FullScreen.min.js" integrity="sha256-6H5xWuqlbGtfk8UL9eMYmp14brCbCw1ZZialT8fHLRE=" crossorigin="anonymous"></script>
</head><body>
<nav class="navbar navbar-inverse">
<div class="container-fluid">
<div class="navbar-header">
<button type="button" class="navbar-toggle" 
data-toggle="collapse" data-target="#myNavbar">
<span class="icon-bar"></span>
<span class="icon-bar"></span>
<span class="icon-bar"></span>
</button>
<a class="navbar-brand" href="#">
PiVPN OpenVPN Status Monitor</a>
</div><div class="collapse navbar-collapse" id="myNavbar">
<ul class="nav navbar-nav"><li class="dropdown">
<a class="dropdown-toggle" data-toggle="dropdown" href="#">VPN
<span class="caret"></span></a>
<ul class="dropdown-menu">
<li><a href="#home_vpn">Home VPN</a></li>
</ul></li>
<li><a href="#map_canvas">Map View</a></li>
</ul>
<a href="#" class="pull-right"><img alt="Logo" 
style="max-height:46px; padding-top:3px;" 
src="images//home/pi/Downloads/openvpn.png"></a>
</div></div></nav>
<div class="container-fluid">
<div class="panel panel-danger" id="home_vpn">
<div class="panel-heading">
<h3 class="panel-title">Home VPN</h3></div>
<div class="panel-body">
Could not connect to 
127.0.0.1:5555 (Connection refused)</div></div>
<div class="panel panel-info"><div class="panel-heading">
<h3 class="panel-title">Map View</h3></div><div class="panel-body">
<div id="map_canvas" style="height:500px"></div>
<script type="text/javascript">
var map = L.map("map_canvas", { fullscreenControl: true, fullscreenControlOptions: { position: "topleft" }  });
var centre = L.latLng(REDACTED);
map.setView(centre, 8);
url = "https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png";
var layer = new L.TileLayer(url, {});
map.addLayer(layer);
var bounds = L.latLngBounds(centre);
var oms = new OverlappingMarkerSpiderfier (map,{keepSpiderfied:true});
var popup = new L.Popup({closeButton:false,offset:new L.Point(0.5,-24)});
oms.addListener("click", function(marker) {
   popup.setContent(marker.alt);
   popup.setLatLng(marker.getLatLng());
   map.openPopup(popup);
});
oms.addListener("spiderfy", function(markers) {
   map.closePopup();
});
map.fitBounds(bounds);
</script>
</div></div>
<div class="well well-sm">
Page automatically reloads every 5 minutes.
Last update: <b>07/04/2020 12:02:52</b></div>
</div></body></html>
DEBUG:
 === begin vpns
{'VPN1': {'connection_lines': 'True',
          'error': 'Connection refused',
          'host': '127.0.0.1',
          'marker': 'True',
          'name': 'Home VPN',
          'port': '5555',
          'show_disconnect': True,
          'socket_connected': False}}
=== end vpns
```
Finally, when I refresh the page at 127.0.0.1 my terminal running the gunicorn server spits out the following:

```
WARNING: Config file does not exist or is unreadable: ./openvpn-monitor.conf
INFO: Using config file: /etc/openvpn-monitor.conf
WARNING: socket error: [Errno 111] Connection refused
```
Thank you in advance for your help!

**Comment by condraj2** [comment]
Closing this out as managed to figure it out with the assistance of a friend.

tl;dr I had two competing versions of OpenVPN running and for whatever reason the version that was being used for OpenVPN Monitor was using the backup_server.conf file rather than the standard server.conf, and as a result, port 5555 wasn't configured to be opened for the management interface.
